source/MaterialXCore/Document.cpp

```diff
@@ -1099,7 +1099,11 @@ void Document::upgradeVersion()
                     InputPtr input = updateChildSubclass<Input>(interface, param);
                     if (interface->isA<NodeDef>())
                     {
-                        input->setIsUniform(true);
+                        // Only strings and filename types should be set as uniforms.
+                        const string& inputType = input->getType();
+                        bool isUniform = (inputType != FILENAME_TYPE_STRING &&
+                                          inputType != STRING_TYPE_STRING);
+                        input->setIsUniform(isUniform);
```

Comment:
One thought here:

We may want to fall back to the default value of this uniform attribute, rather than manually setting "false" for the most common case.

Reply:
Good catch. Should be only has needed now.